libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/MapboxNavigationView.kt

```diff
@@ -0,0 +1,73 @@
+package com.mapbox.navigation.dropin
+
+import com.mapbox.maps.MapView
+import com.mapbox.navigation.core.arrival.ArrivalObserver
+import com.mapbox.navigation.core.directions.session.RoutesObserver
+import com.mapbox.navigation.core.trip.session.BannerInstructionsObserver
+import com.mapbox.navigation.core.trip.session.LocationObserver
+import com.mapbox.navigation.core.trip.session.RouteProgressObserver
+import com.mapbox.navigation.core.trip.session.TripSessionStateObserver
+
+interface MapboxNavigationView {
```

Comment:
As per our previous discussion we want to stay away from interfaces since it is easy to break Semver with this design. Is the plan to introduce versions to scale this in the future? Not a blocker, but I think it would be better to expose these APIs outside of `NavigationView` in some wrapper rather than putting everything in the `NavigationView` which I envision would keep on growing with addition to new functions in this interface.

Reply:
"contract" isn't a conventional name so I don't think we should use it. Lukas an I talked at length about this interface approach. I initially had a versioning strategy. We settled on the approach that's currently in place. 

The interface allows us to keep the `NavigationView` class as black box and expose an API via the interface. We can add to the interface without breaking semver and deprecate as necessary.  We can think of `NavigationView` as a UI component a developer adds to their layout while `MapboxNavigationView` is the API for interacting with the system we often refer to as the drop-in UI.  So `MapboxNavigationView` isn't the UI element it's the API.

Reply:
I suggested `contract` because this is a widely used naming convention in MVP architecture with the exact pattern you have defined here and Android developers are familiar with it. 
I understand that the APIs are being defined here, but I don't like the fact that `NavigationView` will be implementing this interface and hence all the APIs will land in that class as overrides. A developer would then have to make use of `navigationView.*` to invoke any API calls. I am trying to think if there is a better way and avoid doing this. Just like we do for `MapboxTripProgressApi` and `MapboxTripProgressView` where we separate the API from the View and have the `View` only one function in most cases i.e. to render the data. This is not a blocker for this PR but I think we should probably see if this can be improved.
Meanwhile it's still very confusing for me to have a `NavigationView` and `MapboxNavigationView`. As you mentioned `MapboxNavigationView` is nothing but API, can we rename it to `MapboxNavigationViewApi` if not `MapboxNavigationViewContract`

Reply:
NavigationView is not implementing the interface. It knows nothing about it.